contracts/TransferSwapper.sol

```diff
@@ -234,6 +207,33 @@ contract TransferSwapper is MessageReceiverApp, Swapper, SigVerifier, FeeOperato
         emit RequestSent(id, transferId, _desc.dstChainId, _amountIn, _tokenIn, _desc.dstTokenOut);
     }
 
+    function _transfer(
+        bytes32 _id,
+        address _dstTransferSwapper,
+        TransferDescription memory _desc,
+        ICodec.SwapDescription[] memory _dstSwaps,
+        uint256 _amount,
+        address _token
+    ) private returns (bytes32 transferId) {
+        uint256 msgFee = msg.value;
```

Comment:
Use `MessageBus.calcFee()` to calculate `msgFee` and check that `msg.value` (subtract `_desc.amountIn` if `nativeIn`) is sufficient.

Then add a function to extract native fee in `FeeOperator` in case users pass in excess fees.

Reply:
This PR is for security improvements. I'll address this in another PR